### PR TITLE
LogUniformLogPrior

### LINKED ISSUE #1498: Add reciprocal distribution prior (LogUniformLogPrior) to help estimate parameters that vary over multiple orders of magnitude
Hi, thanks for a great package! It would be great to have a  [reciprocal/log uniform](https://en.wikipedia.org/wiki/Reciprocal_distribution) prior. It should be straightforward since it's implemented in `scipy.stats`, and I would be happy to implement it.

### PR DESCRIPTION
**Pull request by abillscmu**
This PR adds a log uniform prior, as implemented in `scipy.stats`. Please let me know if there are any issues or anything, I went through `CONTRIBUTING.md` and think I got it all.

Fixes #1498

### CODE REVIEW

File: docs/source/log_priors.rst [hunk 1]
```diff
@@ -28,6 +28,7 @@ Overview:
 - :class:`StudentTLogPrior`
 - :class:`TruncatedGaussianLogPrior`
 - :class:`UniformLogPrior`
+- :class:`LogUniformLogPrior`
```
Comment (ben18785) [inline docs/source/log_priors.rst:31]: Could we move this up to be in alphabetical order?

File: docs/source/log_priors.rst [hunk 2]
```diff
@@ -57,3 +58,5 @@ Overview:
 .. autoclass:: TruncatedGaussianLogPrior
 
 .. autoclass:: UniformLogPrior
+
+.. autoclass:: LogUniformLogPrior
```
Comment (ben18785) [inline docs/source/log_priors.rst:62]: Same here please

File: pints/__init__.py [hunk 3]
```diff
@@ -105,6 +105,7 @@ def version(formatted=False):
     StudentTLogPrior,
     TruncatedGaussianLogPrior,
     UniformLogPrior,
+    LogUniformLogPrior
```
Comment (ben18785) [inline pints/__init__.py:108]: And here (sorry)

**Review by ben18785 — CHANGES_REQUESTED** [review]
Thanks @abillscmu -- looks really great. Just a few small things here. After looking at the inline comments, would you be able to move the code corresponding to LogUniformPrior so that the class appears in alphabetical order in the `_log_priors.py` file?

File: pints/_log_priors.py [hunk 4]
```diff
@@ -1344,3 +1344,73 @@ def n_parameters(self):
...
+        return scipy.stats.loguniform.cdf(x, self._a, self._b)
+
+    def icdf(self, p):
+        """ See :meth:`LogPrior.icdf()`. """
+        return scipy.stats.loguniform.ppf(p, self._a, self._b)
+
+    def evaluateS1(self, x):
+        """ See :meth:`LogPrior.evaluateS1()`. """
+        dpdfdx = - self._c * np.power(x, - 2)
+        dlogdx = 1 / scipy.stats.loguniform.pdf(x, self._a, self._b)
```
[the remark this answers is not in the document]
Comment (abillscmu) [reply pints/_log_priors.py:1400]: lol, yes, thanks

**Comment by abillscmu** [comment]
Done, thanks for the review, let me know if there's anything else!

**Pull request merged**